src/Orchard.Web/Modules/Orchard.Autoroute/Migrations.cs

```diff
@@ -37,5 +52,23 @@ public class Migrations : DataMigrationImpl {
 
             return 3;
         }
+
+        public int UpdateFrom3() {
+
+            SchemaBuilder.AlterTable("AutoroutePartRecord", table => table
+                .AddColumn<bool>("UseCulturePattern", c => c.WithDefault(false))
+            );
+
+            // populate the AutorouteSettings.DefaultPatternDefinitions AutoroutePart settings
+            foreach (var autoroutePart in _contentManager.Query<AutoroutePart, AutoroutePartRecord>().List()) {
+                var settings = autoroutePart.Settings.GetModel<AutorouteSettings>();
+                string patternIndex = autoroutePart.Settings["AutorouteSettings.DefaultPatternIndex"];
+                settings.DefaultPatterns = new List<DefaultPattern> { new DefaultPattern { PatternIndex = patternIndex, Culture = "" } };

```

Comment:
Not even sure you need to update the value as you can do it while the object is used the first time.